### PR TITLE
Resolve line-ending changes (fix #9)

### LINKED ISSUE #9: Resolve trivial whitespace conflicts
Using resolve-trivial-conflicts, I got confused as to why my conflicts which I trivialized were not resolved. Then I found that it was a matter of different line-ending characters..

```
$ resolve-trivial-conflicts 
examples/audio plugin demo/Source/PluginProcessor.h: Failed to resolve any of the 1 conflicts
$ unix2dos "examples/audio plugin demo/Source/PluginProcessor.h"
unix2dos: converting file examples/audio plugin demo/Source/PluginProcessor.h to DOS format...
$ resolve-trivial-conflicts 
examples/audio plugin demo/Source/PluginProcessor.h: Successfully resolved 1 conflicts (failed to resolve 0 conflicts), git adding
```

### PR DESCRIPTION
**Pull request by yairchu**
On the issue (5 years ago!) I was originally against doing this by default, but I changed my mind,
it's a nice surprise to have your conflicts just resolved :)

* If both A and B have same line ending, apply this edit to base too
* If one side and base both consistently have the same endings and it changes, change in all
* Otherwise (mixed or unclear) preserve the original line endings

### CODE REVIEW

File: src/Conflict.hs [hunk 1]
```diff
@@ -45,6 +45,10 @@ setBodies f = runIdentity . bodies (Identity . f)
 setBodyStrings :: (String -> String) -> Conflict -> Conflict
 setBodyStrings f = runIdentity . bodyStrings (Identity . f)
 
+-- fold
+getBodies :: Conflict -> [[String]]
```
Comment (EyalIO) [inline src/Conflict.hs:49]: Would be nicer to return a traversable of 3 a's?

File: src/Resolution.hs [hunk 2]
```diff
@@ -87,14 +87,58 @@ untabifyStr size =
...
+    where
+        f Mixed _ = Mixed
+        f x c | x == c = x
+            | otherwise = Mixed
+
+allSame :: Eq a => [a] -> Bool
+allSame (x:y:rest) = x == y && allSame (y:rest)
+allSame _ = True
+
+chooseEnding :: [LineEnding] -> LineEnding
```
Comment (EyalIO) [inline src/Resolution.hs:110]: Use a traversable of 3? Perhaps even one representing base,a,b that has this choice function for it? We need it in more places

**Review by EyalIO — APPROVED** [review]

File: src/Conflict.hs (hunk 1 above)
Comment (yairchu) [reply src/Conflict.hs]: Done, changed `Conflict` to hold the data in `Sides` structures and applied this to other code using it as well.

File: src/Resolution.hs (hunk 2 above)
Comment (yairchu) [reply src/Resolution.hs]: Done

**Pull request merged**